storage/store.go

```diff
@@ -1452,7 +1452,40 @@ func (s *store) canUseShifting(uidmap, gidmap []idtools.IDMap) bool {
 // On entry:
 // - rlstore must be locked for writing
 // - rlstores MUST NOT be locked
-func (s *store) putLayer(rlstore rwLayerStore, rlstores []roLayerStore, id, parent string, names []string, mountLabel string, writeable bool, lOptions *LayerOptions, diff io.Reader, slo *stagedLayerOptions) (*Layer, int64, error) {
+func (s *store) putLayer(rlstore rwLayerStore, id string, parentLayer *Layer, names []string, mountLabel string, writeable bool, options *LayerOptions, slo *stagedLayerOptions) (*Layer, int64, error) {
+	return rlstore.create(id, parentLayer, names, mountLabel, nil, options, writeable, slo)
+}
+
+// On entry:
+// - rlstore must be locked for writing
+// - rlstores MUST NOT be locked
+func getParentLayer(rlstore rwLayerStore, rlstores []roLayerStore, parent string) (*Layer, error) {
+	var ilayer *Layer
+	for _, l := range append([]roLayerStore{rlstore}, rlstores...) {
+		lstore := l
+		if lstore != rlstore {
+			if err := lstore.startReading(); err != nil {
+				return nil, err
+			}
+			defer lstore.stopReading()
+		}
+		if l, err := lstore.Get(parent); err == nil && l != nil {
+			ilayer = l
```

Comment:
(As is, this could `return l` immediately`, but first see elsewhere about keeping the parent store locked.)